spec/predicates/release.md

```diff
@@ -0,0 +1,98 @@
+# Predicate type: Release
+
+Type URI: https://in-toto.io/attestation/release
+
+Version 0.1
+
+## Purpose
+
+To authoritatively link from a specific release name and version string in a registry, to the artifact names and hashes that make up that release.
+
+## Use Cases
+
+When receiving a new release version, package registries can publish a release attestation covering the artifact names and hashes that make up that release. This allows consumers of that release version to ensure the artifacts they are consuming have not been tampered with.
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
+
+## Schema
+
+### Fields
```

Comment:
suggestion: Should we add an optional timestamp indicating when the package was released?

Reply:
I'm on the fence, but leaning towards no. It's unlikely a bare in-toto attestation will be hanging around; they should be sent to some transparency log or otherwise made tamper-resistant.

In the case of npm publish attestations they are published to the public-good Sigstore Rekor, which of course includes a witnessing timestamp. Does the in-toto attestation itself need a timestamp? Can you describe the use-case where the attestation itself needs the timestamp?

Reply:
> If the intent is to always put this on a transparency log, should we instead be looking toward generalizing the go checksum database rather than creating an attestation type

The way I view this is the release attestation allows registries the benefit / possibility of using a shared transparency log (like Sigstore public-good Rekor) without each registry having to maintain their own transparency log.

If registries want to maintain a transparency log - great! If we want to make it easier for registries to maintain their own transparency log - also great! But today it's a heavy lift for a registry to maintain their own transparency log.

In a similar vein, https://github.com/repository-service-tuf/repository-service-tuf is working towards making it easier for repositories to maintain their own TUF root that signs the repository index metadata. That project has a similar goal to release attestations - securely distributing release artifact hashes - but goes about it in a very different way.